src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -380,15 +380,16 @@ public override void DidReceiveChallenge (NSUrlSession session, NSUrlSessionTask
 				}
 
 				if (challenge.ProtectionSpace.AuthenticationMethod == NSUrlProtectionSpace.AuthenticationMethodNTLM) {
+					NSUrlCredential credential;
 					if (sessionHandler.Credentials != null) {
 						var credentialsToUse = sessionHandler.Credentials as NetworkCredential;
 						if (credentialsToUse == null) {
 							var uri = inflight.Request.RequestUri;
 							credentialsToUse = sessionHandler.Credentials.GetCredential (uri, "NTLM");
 						}
-						var credential = new NSUrlCredential (credentialsToUse.UserName, credentialsToUse.Password, NSUrlCredentialPersistence.ForSession);
-						completionHandler (NSUrlSessionAuthChallengeDisposition.UseCredential, credential);
+						credential = new NSUrlCredential (credentialsToUse.UserName, credentialsToUse.Password, NSUrlCredentialPersistence.ForSession);
 					}
+					completionHandler (NSUrlSessionAuthChallengeDisposition.UseCredential, credential ?? challenge.ProposedCredential);
```

Comment:
It seems weird using `.UseCredential` without credentials - please double check docs

Still the current code is hits `return` without calling the `completionHandler` so that's more than likely wrong...

Could it be possible the return statement should be moved into the `if (sessionHandler.Credentials != null) {` condition ?

That would solve both inconsistency, since `completionHandler` would be called (it's not now) and asked to `.PerformDefaultHandling` with the challenge proposed credentials.